### PR TITLE
[zh]translate node-pressure-eviction.md

### PR DESCRIPTION
**Pull request by howieyuen**

### CODE REVIEW

**Comment by jihoon-seo** [comment]
Hello @howieyuen :)

The [Netlify deploy log](https://app.netlify.com/sites/kubernetes-io-master-staging/deploys/60c333fa48664b0007737805) says that:

> ERROR 2021/06/11 10:00:30 [zh] "docs/concepts/scheduling-eviction/node-pressure-eviction.md": "node-pressure-eviction" is not a valid glossary term_id, see ./docs/reference/glossary/* for a full list

I recommend you to do:

- Copy `content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md` 
to `content/zh/docs/concepts/scheduling-eviction/`
- And translate `content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md` into Chinese

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 1]
```diff
@@ -0,0 +1,777 @@
...
+the kubelet respects your configured `eviction-max-pod-grace-period`. If you use
+[hard eviction thresholds](#hard-eviction-thresholds), it uses a `0s` grace period for termination.
+
+If the pods are managed by a {{< glossary_tooltip text="workload" term_id="workload" >}}
+resource (such as {{< glossary_tooltip text="StatefulSet" term_id="statefulset" >}}
+or {{< glossary_tooltip text="Deployment" term_id="deployment" >}}) that
+replaces failed pods, the control plane or `kube-controller-manager` creates new 
+pods in place of the evicted pods.
+-->
+kubelet 并不尊重你配置的 `PodDisruptionBudget` 或者是 Pod 的 `terminationGracePeriodSeconds`。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:48]:
```suggestion
kubelet 并不理会你配置的 `PodDisruptionBudget` 或者是 Pod 的 `terminationGracePeriodSeconds`。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+如果你使用了[软驱逐阈值](#soft-eviction-thresholds)，kubelet 会尊重你配置的 `eviction-max-pod-grace-period`。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:49]: respect 这个词不太好处理，可以根据上下文翻译成合乎汉语表达习惯的词。
直译为“尊重”有点生硬。

```suggestion
如果你使用了[软驱逐阈值](#soft-eviction-thresholds)，kubelet 会考虑你所配置的
`eviction-max-pod-grace-period`。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 3]
```diff
@@ -0,0 +1,777 @@
...
+或者 {{< glossary_tooltip text="Deployment" term_id="deployment" >}}）管理，
+则控制平面或 `kube-controller-manager` 会创建新的 Pod 来代替被驱逐的 Pod。
+
+{{<note>}}
+<!-- 
+The kubelet attempts to [reclaim node-level resources](#reclaim-node-resources)
+before it terminates end-user pods. For example, it removes unused container
+images when disk resources are starved.
+-->
+kubelet 在终止最终用户 Pod 之前尝试[回收节点级资源](#reclaim-node-resources)。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:63]:
```suggestion
kubelet 在终止最终用户的 Pod 之前会尝试[回收节点级资源](#reclaim-node-resources)。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+| `pid.available`      | `pid.available` := `node.stats.rlimit.maxpid` - `node.stats.rlimit.curproc`           |
+
+<!-- 
+In this table, the `Description` column shows how kubelet gets the value of the
+signal. Each signal supports either a percentage or a literal value. Kubelet 
+calculates the percentage value relative to the total capacity associated with
+the signal. 
+-->
+在上表中，`描述`列显示了 kubelet 如何获取信号的值。每个信号支持百分比值或者是字面值。
+kubelet 计算相对于与信号有关的总容量的百分比值。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:114]:
```suggestion
kubelet 计算相对于与信号有关的总量的百分比值。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+
+Unlike pod eviction, if a container is OOM killed, the `kubelet` can restart it 
+based on its `RestartPolicy`.
+-->
+如果 kubelet 在节点遇到 OOM 之前无法回收内存，
+则 `oom_killer` 根据它在节点上使用的内存百分比计算 `oom_score`，
+然后加上 `oom_score_adj` 得到每个容器有效的 `oom_score`。
+然后它会杀死得分最高的容器。
+
+这意味着低 QoS Pod 中的容器相对于其调度请求消耗大量内存，将首先被杀死。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:613]:
```suggestion
这意味着低 QoS Pod 中相对于其调度请求消耗内存较多的容器，将首先被杀死。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 6]
```diff
@@ -0,0 +1,777 @@
...
+<!-- 
+#### Schedulable resources and eviction policies
+
+When you configure the kubelet with an eviction policy, you should make sure that
+the scheduler will not schedule pods if they will trigger eviction because they
+immediately induce memory pressure.
+-->
+#### 可调度的资源和驱逐策略
+
+当你使用驱逐策略配置 kubelet 时，
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:636]:
```suggestion
当你为 kubelet 配置驱逐策略时，
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 7]
```diff
@@ -0,0 +1,777 @@
...
+你应该确保调度程序不会在 Pod 触发驱逐时对其进行调度，因为它们会立即引起内存压力。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:637]:
```suggestion
你应该确保调度程序不会在 Pod 触发驱逐时对其进行调度，因为这类 Pods 会立即引起内存压力。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 8]
```diff
@@ -0,0 +1,777 @@
...
+
+* Node memory capacity: `10Gi`
+* Operator wants to reserve 10% of memory capacity for system daemons (kernel, `kubelet`, etc.)
+* Operator wants to evict Pods at 95% memory utilization to reduce incidence of system OOM.
+-->
+考虑以下场景：
+
+* 节点内存容量：`10Gi`
+* 操作员希望为系统守护进程（内核、`kubelet` 等）保留 10% 的内存容量
+* 操作员希望以 95% 的内存利用率驱逐 Pod，以减少系统 OOM 的发生率。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:650]:
```suggestion
* 操作员希望驱逐内存利用率为 95% 的Pod，以减少系统 OOM 的概率。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 9]
```diff
@@ -0,0 +1,777 @@
...
+
+<!-- 
+For this to work, the kubelet is launched as follows:
+-->
+为此，kubelet 启动如下：
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:655]:
```suggestion
为此，kubelet 启动设置如下：
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 10]
```diff
@@ -0,0 +1,777 @@
...
+for the system, which is `10% of the total memory + the eviction threshold amount`. 
+
+The node can reach the eviction threshold if a pod is using more than its request,
+or if the system is using more than `1Gi` of memory, which makes the `memory.available`
+signal fall below `500Mi` and triggers the threshold. 
+-->
+在此配置中，`--system-reserved` 标志为系统预留了 `1.5Gi` 的内存，
+即 `总内存的 10% + 驱逐阈值量`。
+
+如果 Pod 使用的内存超过其请求，或者系统使用的内存超过 `1Gi`，
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:673]:
```suggestion
如果 Pod 使用的内存超过其请求值或者系统使用的内存超过 `1Gi`，
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 11]
```diff
@@ -0,0 +1,777 @@
...
+
+By default, the kubelet polls `cAdvisor` to collect memory usage stats at a
+regular interval. If memory usage increases within that window rapidly, the
+kubelet may not observe `MemoryPressure` fast enough, and the `OOMKiller`
+will still be invoked. 
+-->
+#### kubelet 可能不会立即观察到内存压力
+
+默认情况下，kubelet 轮询 `cAdvisor` 以定期收集内存使用情况统计信息。
+如果该窗口内的内存使用量迅速增加，kubelet 可能无法足够快地观察到 `MemoryPressure`，
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:713]:
```suggestion
如果该轮询时间窗口内内存使用量迅速增加，kubelet 可能无法足够快地观察到 `MemoryPressure`，
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 12]
```diff
@@ -0,0 +1,777 @@
...
+并且 `OOMKiller` 仍将被调用。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:714]:
```suggestion
但是 `OOMKiller` 仍将被调用。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 13]
```diff
@@ -0,0 +1,777 @@
...
+is crossed.
+
+If you are not trying to achieve extreme utilization, but a sensible measure of
+overcommit, a viable workaround for this issue is to use the `--kube-reserved`
+and `--system-reserved` flags to allocate memory for the system. 
+-->
+你可以使用 `--kernel-memcg-notification`
+标志在 kubelet 上启用 `memcg` 通知 API，以便在超过阈值时立即收到通知。
+
+如果你不是要达到极端利用率，而是要采取合理的过量使用措施，
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:728]:
```suggestion
如果你不是追求极端利用率，而是要采取合理的过量使用措施，
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 14]
```diff
@@ -0,0 +1,777 @@
...
+On Linux, the kernel tracks the number of bytes of file-backed memory on active 
+LRU list as the `active_file` statistic. The kubelet treats `active_file` memory
+areas as not reclaimable. For workloads that make intensive use of block-backed 
+local storage, including ephemeral local storage, kernel-level caches of file 
+and block data means that many recently accessed cache pages are likely to be 
+counted as `active_file`. If enough of these kernel block buffers are on the 
+active LRU list, the kubelet is liable to observe this as high resource use and 
+taint the node as experiencing memory pressure - triggering pod eviction.
+-->
+#### active_file 内存不视为可用内存
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:743]:
```suggestion
#### active_file 内存未被视为可用内存
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 15]
```diff
@@ -0,0 +1,777 @@
...
+
+在 Linux 上，内核跟踪活动 LRU 列表上的文件支持内存的字节数作为 `active_file` 统计信息。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:745]:
```suggestion
在 Linux 上，内核跟踪活动 LRU 列表上的基于文件所虚拟的内存字节数作为 `active_file` 统计信息。
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 16]
```diff
@@ -0,0 +1,777 @@
...
+kubelet 将 `active_file` 内存区域视为不可回收。
+对于大量使用块支持的本地存储的工作负载，包括临时本地存储，
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:747]:
```suggestion
对于大量使用块设备形式的本地存储（包括临时本地存储）的工作负载，
```

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 17]
```diff
@@ -0,0 +1,777 @@
...
+文件和块数据的内核级缓存意味着许多最近访问的缓存页面可能被计为 `active_file`。
+如果这些内核块缓冲区中在活动 LRU 列表上有足够多，
+kubelet 很容易将其视为高资源使用并将节点污染为经历内存压力——触发 pod 驱逐。
```
Comment (tengqm) [inline content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:750]:
```suggestion
kubelet 很容易将其视为资源用量过量并为节点设置内存压力污点，从而触发 Pod 驱逐。
```

**Comment by howieyuen** [comment]
> Hello @howieyuen :)
> 
> The [Netlify deploy log](https://app.netlify.com/sites/kubernetes-io-master-staging/deploys/60c333fa48664b0007737805) says that:
> 
> > ERROR 2021/06/11 10:00:30 [zh] "docs/concepts/scheduling-eviction/node-pressure-eviction.md": "node-pressure-eviction" is not a valid glossary term_id, see ./docs/reference/glossary/* for a full list
> 
> I recommend you to do:
> 
> * Copy `content/en/docs/reference/glossary/node-pressure-eviction.md`
>   to `content/zh/docs/reference/glossary/`
> * And translate `content/zh/docs/reference/glossary/node-pressure-eviction.md` into Chinese

thanks for the tips

File: content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 18]
```diff
@@ -0,0 +1,777 @@
...
+A hard eviction threshold has no grace period. When a hard eviction threshold is
+met, the kubelet kills pods immediately without graceful termination to reclaim
+the starved resource.
+
+You can use the `eviction-hard` flag to configure a set of hard eviction 
+thresholds like `memory.available<1Gi`. 
+-->
+#### 硬驱逐阈值 {#hard-eviction-thresholds}
+
+硬驱逐阈值没有宽限期。当达到硬驱逐阈值时，
```
[the remark this answers is not in the document]
Comment (howieyuen) [reply content/zh/docs/concepts/scheduling-eviction/node-pressure-eviction.md:263]: emm…I agree

**Comment by jihoon-seo** [comment]
After adding lgtm comment, I thought that 
it is not appropriate for me (who cannot review Chinese contribution) to add lgtm.
So I deleted my lgtm comment, but the bot already merged it.
Sorry for causing confusion.

**Pull request merged**